keel-titus-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/titus/cluster/TitusClusterHandler.kt

```diff
@@ -521,19 +521,28 @@ class TitusClusterHandler(
   private suspend fun getExistingServerGroupsByRegion(resource: Resource<TitusClusterSpec>): Map<String, List<ClouddriverTitusServerGroup>> {
     val existingServerGroups: MutableMap<String, MutableList<ClouddriverTitusServerGroup>> = mutableMapOf()
 
-    cloudDriverService
-      .listTitusServerGroups(
-        user = resource.serviceAccount,
-        app = resource.spec.application,
-        account = resource.spec.locations.account,
-        cluster = resource.spec.moniker.toString()
-      )
-      .serverGroups
-      .forEach { sg ->
-        val existing = existingServerGroups.getOrPut(sg.region, { mutableListOf() })
-        existing.add(sg)
-        existingServerGroups[sg.region] = existing
+    try {
+      cloudDriverService
+        .listTitusServerGroups(
+          user = resource.serviceAccount,
+          app = resource.spec.application,
+          account = resource.spec.locations.account,
+          cluster = resource.spec.moniker.toString()
+        )
+        .serverGroups
+        .forEach { sg ->
+          val existing = existingServerGroups.getOrPut(sg.region, { mutableListOf() })
+          existing.add(sg)
+          existingServerGroups[sg.region] = existing
+        }
+    } catch (e: HttpException) {
```

Comment:
should we log the error here?

Reply:
We don't need to log since if it's a 404 we expect that (well, we NOW expect it), and otherwise we throw the exception